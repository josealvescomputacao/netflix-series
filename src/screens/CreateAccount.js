import React, {Component, Fragment} from 'react'
import  {Redirect} from 'react-router-dom'
import { connect } from 'react-redux'
import ActionCreator from '../redux/actionCreators'
import styled from 'styled-components'

import FormAccount from './elements/formAccount'

const BodyRegister = styled.div`

    .positionMessage{
        position: absolute
        top: 350px
        width: 100%
    }
`

export class ScreensCreateAccount extends Component {    
    
    state={ 
        name: "",
        email: "",
        password: "",
        passwordConfirm: "",  
    }
    
    componentDidMount(){
        this.props.reset() 
    }

    
    handleChange = field => event => {
        const form = {
            ...this.state,
            [field]: event.target.value
        }
        if (field === 'passwordConfirm'){
            if (form.password.length < 6){
                this.setState({password:'', passwordConfirm: ''})
                this.props.error('The password needs has to more than 6 characters!')  
                return
            }  
            if (form[field].length >= form.password.length && form[field] !== form.password){
                this.setState({passwordConfirm: ''})
                this.props.error('The password confirm is wrong')
                return
            }  
        }
        document.getElementById('password').onpaste = () => false
        document.getElementById('passwordConfirm').onpaste = () => false
        this.setState({...form})
    }
    

    createAccount = () => { 
        if (this.state.name.length === 0){ 
            this.props.error('Write your name!')
            return           
        }else if (this.state.name.length > 0 && this.state.name.length < 3){
            this.props.error('The name needs has to more than 2 characters!')
            return           
        }
        if (this.state.passwordConfirm !== this.state.password){
            this.setState({passwordConfirm: ''})
            this.props.error('The password confirm  is wrong')
            return
        }
        const user = {
            name : this.state.name, 
            email : this.state.email, 
            passwd : this.state.passwordConfirm
        }
        this.props.createAccount(user)          
    }   

   
    render(){
        return (
            <Fragment>  
                {this.props.auth.isAuth && <Redirect to='/'/>} 
                <BodyRegister>
                    {this.props.auth.isLoadding && 
                        <p className='text-info positionMessage'>Is registring...</p>
                    }
                    {this.props.auth.error && 
                        <p className="text-danger positionMessage">{this.props.auth.errorMessage}</p>
                    }
                    <div>
                        <FormAccount handleChange={this.handleChange} createAccount={this.createAccount} state={this.state}/>  
                    </div>
                </BodyRegister>
            </Fragment>
        )   
    }
    
}

const mapStateToProps = state => {
    return {
        auth : state.auth
    }
}
const mapDispatchToProps = dispatch => { 
    return {
        createAccount : user => dispatch(ActionCreator.createProfileRequest(user)),
        error : message => dispatch(ActionCreator.createProfileFailure(message)),
        reset : () => dispatch(ActionCreator.resetAuth())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ScreensCreateAccount) 