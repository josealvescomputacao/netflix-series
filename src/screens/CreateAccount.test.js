import React from 'react'

import {ScreensCreateAccount} from './CreateAccount'

describe('Testing CreateAccount Component', () => {
    const props = {
        reset: jest.fn(),
        auth: {
            isAuth: false,
            isLoadding: false
        }
    }
    it('should render correctly with props and empty state', () => {
        const wrapper = shallow(<ScreensCreateAccount {...props}/>)
        
        expect(wrapper.state().name).toEqual('')
        expect(wrapper.state().email).toEqual('')
        expect(wrapper.state().password).toEqual('')
        expect(wrapper.state().passwordConfirm).toEqual('')
        
        expect(wrapper).toMatchSnapshot()
    })
})    


