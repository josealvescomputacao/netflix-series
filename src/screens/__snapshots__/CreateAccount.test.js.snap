// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CreateAccount Component should render correctly with props and empty state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScreensCreateAccount
    auth={
      Object {
        "isAuth": false,
        "isLoadding": false,
      }
    }
    reset={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        <styled.div>
          <div>
            <FormAccount
              createAccount={[Function]}
              handleChange={[Function]}
              state={
                Object {
                  "email": "",
                  "name": "",
                  "password": "",
                  "passwordConfirm": "",
                }
              }
            />
          </div>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            false,
            undefined,
            <div>
              <FormAccount
                createAccount={[Function]}
                handleChange={[Function]}
                state={
                  Object {
                    "email": "",
                    "name": "",
                    "password": "",
                    "passwordConfirm": "",
                  }
                }
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormAccount
                createAccount={[Function]}
                handleChange={[Function]}
                state={
                  Object {
                    "email": "",
                    "name": "",
                    "password": "",
                    "passwordConfirm": "",
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "createAccount": [Function],
                "handleChange": [Function],
                "state": Object {
                  "email": "",
                  "name": "",
                  "password": "",
                  "passwordConfirm": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          <styled.div>
            <div>
              <FormAccount
                createAccount={[Function]}
                handleChange={[Function]}
                state={
                  Object {
                    "email": "",
                    "name": "",
                    "password": "",
                    "passwordConfirm": "",
                  }
                }
              />
            </div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              false,
              undefined,
              <div>
                <FormAccount
                  createAccount={[Function]}
                  handleChange={[Function]}
                  state={
                    Object {
                      "email": "",
                      "name": "",
                      "password": "",
                      "passwordConfirm": "",
                    }
                  }
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormAccount
                  createAccount={[Function]}
                  handleChange={[Function]}
                  state={
                    Object {
                      "email": "",
                      "name": "",
                      "password": "",
                      "passwordConfirm": "",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "createAccount": [Function],
                  "handleChange": [Function],
                  "state": Object {
                    "email": "",
                    "name": "",
                    "password": "",
                    "passwordConfirm": "",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
